models/user.model.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+
+const userSchema = new mongoose.Schema({
+    username: {
+        type: String,
+        required: "Username is a mandatory attribute",
+        unique: true
```

Comment:
Making username as unique is not a good idea. Because there would be many people with same name

Reply:
This username would be similar to twitter/instagram's username which is unique. I am also capturing the name field which is not unique.